### PR TITLE
[353] Use a privileged action for iterating over the service loader w…

### LINKED ISSUE #353: RestClientBuilderResolver does not work when the security manager is enabled
The `RestClientBuilderResolver` does some privileged actions for various things like looking up the class loader. However, the `loadSpi()` uses a `ServiceLoader` and does not use a privileged action requiring ` java.io.FilePermission` to be added to read the SPI JAR.

### PR DESCRIPTION
**Pull request by jamezp**
…hen locating the RestClientBuilderResolver SPI.

resolves #353 

It would be nice if we could get this in 3.0 as well. If that seems acceptable, I'm happy to open up another PR.

### CODE REVIEW

File: api/src/main/java/org/eclipse/microprofile/rest/client/spi/RestClientBuilderResolver.java [hunk 1]
```diff
@@ -95,19 +95,21 @@ private static RestClientBuilderResolver loadSpi(ClassLoader cl) {
...
+                }
+                instance = spi;
             }
-            instance = spi;
-        }
+            return instance;
+        };
 
-        return instance;
+        return System.getSecurityManager() == null ? action.run() : AccessController.doPrivileged(action);
```
Comment (rdebusscher) [inline api/src/main/java/org/eclipse/microprofile/rest/client/spi/RestClientBuilderResolver.java:112]: It is not wise to introduce a new usage of SecurityManager which is Deprecated for removal in JDK 18. This limits the usability in an LTS JDK21+.

**Review by rdebusscher — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/eclipse/microprofile/rest/client/spi/RestClientBuilderResolver.java (hunk 1 above)
Comment (jamezp) [reply api/src/main/java/org/eclipse/microprofile/rest/client/spi/RestClientBuilderResolver.java:112]: There are users that require the security manager and this already has references to the security manager. It was just missed in this one spot.

Comment (Emily-Jiang) [reply api/src/main/java/org/eclipse/microprofile/rest/client/spi/RestClientBuilderResolver.java:112]: We can merge this one and then clean up all SecurityManager usage when it is removed from JDK in due course.

**Review by Emily-Jiang — APPROVED** [review]
LGTM

**Pull request merged**